zerver/tornado/event_queue.py

```diff
@@ -937,6 +939,10 @@ def process_presence_event(event: Mapping[str, Any], users: Iterable[int]) -> No
         presence=event['presence'],
     )
 
+    if "instrumented_event_id" in event:
```

Comment:
@timabbott Apart from the changes you requested, I made this change as well since the tests were failing.